src/esbmc/bmc.cpp

```diff
@@ -606,10 +606,28 @@ smt_convt::resultt bmct::run_thread(std::shared_ptr<symex_target_equationt> &eq)
   {
     fine_timet slice_start = current_time();
     BigInt ignored;
-    if(!options.get_bool_option("no-slice"))
-      ignored = slice(eq, options.get_bool_option("slice-assumes"));
-    else
+    if(options.get_bool_option("no-slice"))
       ignored = simple_slice(eq);
+    else
+    {
+      struct
+      {
+        const std::unordered_set<std::string> &names, &ids;
+
+        bool operator()(const symbol2t &sym) const
+        {
+          return names.count(sym.thename.as_string()) ||
+                 ids.count(sym.get_symbol_name());
+        }
+      } no_slice_symbol = {
+        config.no_slice_names,
+        config.no_slice_ids,
+      };
+
+      ignored =
+        slice(eq, options.get_bool_option("slice-assumes"), no_slice_symbol);
```

Comment:
Can you pass the options object and get both the `slice-assumes` flag and the list inside the `symex_slicet` constructor?

Reply:
Or move this block into a separate function. Or to inside `config.h`.

Reply:
Well, in that case I could get the `"no-slice"` option as well to distinguish between slice() and simple_slice() and don't need the `no_slice` callback as a member. I've added a commit that does this and also moved the helper class `symex_slicet` into an anon namespace in the .cpp file.